plugins/versioning/src/main/kotlin/versioning/VersionsNavigationCreator.kt

```diff
@@ -1,53 +1,86 @@
 package org.jetbrains.dokka.versioning
 
+import com.jetbrains.rd.util.first
 import kotlinx.html.a
-import kotlinx.html.button
 import kotlinx.html.div
-import kotlinx.html.i
 import kotlinx.html.stream.appendHTML
+import org.jetbrains.dokka.base.renderers.html.strike
 import org.jetbrains.dokka.plugability.DokkaContext
+import org.jetbrains.dokka.plugability.configuration
 import org.jetbrains.dokka.plugability.plugin
 import org.jetbrains.dokka.plugability.querySingle
 import java.io.File
-import java.nio.file.Files.isDirectory
-import java.nio.file.Path
 
 interface VersionsNavigationCreator {
-    operator fun invoke(): String
+    operator fun invoke(location: String): String
```

Comment:
i don't like this as it is too general for what you are doing therefore doesn't let the reader know that this is in fact a relative location from the current version. 
If you really want to pass the location always (either in string or file format) why don't we just stick to files?